code/modules/surgery/organs/stomach/external/wings.dm

```diff
@@ -0,0 +1,203 @@
+
+///The true wings that you can use to fly and shit (you cant actually shit with them, but it does wing stuff)
+/obj/item/organ/external/wings/functional
+	///The flight action object
+	var/datum/action/innate/flight/fly
+
+	///The preference type for opened wings
+	var/wings_open_preference = "wingsopen"
+	///The preference type for closed wings
+	var/wings_closed_preference = "wings"
+
+	///Are our wings open or closed?
+	var/wings_open = FALSE
+
+/obj/item/organ/external/wings/functional/get_global_feature_list()
+	if(wings_open)
+		return GLOB.wings_open_list
+	else
+		return GLOB.wings_list
+
+/obj/item/organ/external/wings/functional/Insert(mob/living/carbon/reciever, special, drop_if_replaced)
+	. = ..()
+
+	if(isnull(fly))
+		fly = new
+		fly.Grant(reciever)
+
+/obj/item/organ/external/wings/functional/Remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+
+	fly.Remove(organ_owner)
+
+/obj/item/organ/external/wings/functional/on_life(delta_time, times_fired)
+	. = ..()
+
+	HandleFlight(owner)
+
+///Mostly as a starter to check if we can still fly
+/obj/item/organ/external/wings/functional/proc/HandleFlight(mob/living/carbon/human/H)
+	if(H.movement_type & FLYING)
+		if(!CanFly(H))
+			ToggleFlight(H)
+			return FALSE
+		return TRUE
+	else
+		return FALSE
+
+///Check if we can still fly
+/obj/item/organ/external/wings/functional/proc/CanFly(mob/living/carbon/human/H)
+	if(H.stat || H.body_position == LYING_DOWN)
+		return FALSE
+	if(H.wear_suit && ((H.wear_suit.flags_inv & HIDEJUMPSUIT) && (!H.wear_suit.species_exception || !is_type_in_list(src, H.wear_suit.species_exception)))) //Jumpsuits have tail holes, so it makes sense they have wing holes too
+		to_chat(H, span_warning("Your suit blocks your wings from extending!"))
+		return FALSE
+	var/turf/T = get_turf(H)
+	if(!T)
+		return FALSE
+
+	var/datum/gas_mixture/environment = T.return_air()
+	if(environment && !(environment.return_pressure() > 30))
+		to_chat(H, span_warning("The atmosphere is too thin for you to fly!"))
+		return FALSE
+	else
+		return TRUE
+
+///Slipping but in the air?
+/obj/item/organ/external/wings/functional/proc/flyslip(mob/living/carbon/human/H)
+	var/obj/buckled_obj
+	if(H.buckled)
+		buckled_obj = H.buckled
+
+	to_chat(H, span_notice("Your wings spazz out and launch you!"))
+
+	playsound(H.loc, 'sound/misc/slip.ogg', 50, TRUE, -3)
+
+	for(var/obj/item/I in H.held_items)
+		H.accident(I)
+
+	var/olddir = H.dir
+
+	H.stop_pulling()
+	if(buckled_obj)
+		buckled_obj.unbuckle_mob(H)
+		step(buckled_obj, olddir)
+	else
+		new /datum/forced_movement(H, get_ranged_target_turf(H, olddir, 4), 1, FALSE, CALLBACK(H, /mob/living/carbon/.proc/spin, 1, 1))
+	return TRUE
+
+///UNSAFE PROC, should only be called through the Activate or other sources that check for CanFly
+/obj/item/organ/external/wings/functional/proc/ToggleFlight(mob/living/carbon/human/H)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

toggle_flight